sc/social/like/upgrades/v3046/__init__.py

```diff
@@ -16,13 +25,24 @@ def reindex_catalog(setup_tool):
     results = catalog()
     logger.info(u'Found {0} objects'.format(len(results)))
     n = 0
+    errors = 0
     for obj in get_valid_objects(results):
-        catalog.catalog_object(obj, idxs=['object_provides'], update_metadata=False)
+        try:
+            catalog.catalog_object(obj, idxs=['object_provides'], update_metadata=False)
+        except ConflictError:
+            raise
+        except Exception as e:
```

Comment:
I don't think this is a good idea; if you don't know what other exception can be raised here, better to remove this also or we can hide some other issues.

Reply:
Wouldn't this operation be better atomic? The only improvement that could be made is to log the object that failed but raise the exception right away. If the upgrade runs without error, many users may think that everything went well when it didn't.